mitodl/mitxonline
# Added enrollment sync when dashboard loads

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #74

#### What's this PR do?
- Syncs a user's local enrollment records with edX every time that user's full enrollment data is requested (if a feature flag is turned on)
- Adds a management command to sync enrollment data

#### How should this be manually tested?
- Add `FEATURE_SYNC_ON_DASHBOARD_LOAD=True` to your `.env`
- Run the app alongside edX, and make sure you have at least 1 course run that exists both locally and in edX
- Load a user's dashboard. It should load successfully
- Enroll that user in a course run that has a matching run in edX
- Navigate to the course on edX and unenroll (should be doable via the gear icon)
- Check your enrollment in mitxonline Django admin. It should still exist and should still be active
- Load the dashboard. You should see that the enrollment no longer shows up in the dashboard, and in Django admin you should see the enrollment set to inactive

You can also test the management command (`sync_enrollments`) by leaving off the last step and running the management command instead.

#### Any background context you want to provide?
- There's a noticeable "blink" when the dashboard data is loading now. We need to settle on a loading UI. Mentioning it here just to say that it's a known issue and is out of scope for this PR

#### Screenshots (if appropriate)
![ss 2021-08-19 at 13 04 11 ](https://user-images.githubusercontent.com/14932219/130132975-ead5f2ba-72c7-40db-82fe-87b403663029.png)
